src/main/java/com/selfxdsd/todos/PuzzlesApi.java

```diff
@@ -89,27 +88,25 @@ public PuzzlesApi(
     )
     public ResponseEntity<String> reviewPuzzles(
         @PathVariable final String provider,
-        @PathVariable final String owner, @PathVariable final String name,
+        @PathVariable final String owner,
+        @PathVariable final String name,
         @RequestBody final String payload
-    ) {
+    ) throws Exception {
+        final Self selfCore = this.beanFactory.getBean(Self.class);
         final ResponseEntity<String> resp;
-        final Project project = this.selfCore.projects().getProjectById(
-            owner + "/" + name, provider
+        final String fullName = owner + "/" + name;
+        final Project project = selfCore.projects().getProjectById(
+            fullName, provider
         );
         if (project == null) {
             resp = ResponseEntity.badRequest().build();
         } else {
-            if (provider.equalsIgnoreCase(Provider.Names.GITHUB)) {
-                this.puzzlesComponent.review(
-                    WebhookEvents.create(project, "push", payload)
-                );
-            } else if (provider.equalsIgnoreCase(Provider.Names.GITLAB)) {
-                this.puzzlesComponent.review(
-                    WebhookEvents.create(project, "Push Hook", payload)
-                );
-            }
+            this.eventPublisher.publishEvent(
+                new PuzzlesWebhookEvent(provider, fullName, payload)
```

Comment:
I is all good, we send a Spring event instead to be processed in different thread (async) in PuzzlesComponent